pkg/endpoint/bpf.go

```diff
@@ -735,8 +735,18 @@ func (e *Endpoint) runPreCompilationSteps(regenContext *regenerationContext, rul
 	stats := &regenContext.Stats
 	datapathRegenCtxt := regenContext.datapathRegenerationContext
 
+	// regenerate policy without holding the lock.
+	// This is because policy generation needs the ipcache to make progress, and the ipcache needs to call
+	// endpoint.ApplyPolicyMapChanges()
+	stats.policyCalculation.Start()
+	policyResult, err := e.regeneratePolicy()
+	stats.policyCalculation.End(err == nil)
```

Comment:
I realize you're probably just propagating this forward from the previous implementation, but inside `e.regeneratePolicy()` it's already accounting `stats.policyCalculation`. It shouldn't be in both places, probably we can remove it from the caller here.

Reply:
Hmm, they're separate statistics - one is a `policyRegenerationStatistics{}` and the other is `regenerationStatistics{}` :-)

Reply:
Oh god.